src/runtime_src/core/common/sensor.cpp

```diff
@@ -211,15 +215,96 @@ read_electrical(const xrt_core::device * device)
   root.put("power_consumption_max_watts", max_power_watts);
   root.put("power_consumption_watts", power_watts);
   root.put("power_consumption_warning", warning);
+
   return root;
 }
 
-
 ptree_type
-read_thermals(const xrt_core::device * device)
+read_electrical(const xrt_core::device * device)
 {
   ptree_type root;
+  ptree_type sensor_array;
+  ptree_type pt;
+  bool is_data_driven = true;
+
+  try {
+    auto current  = xrt_core::device_query<xq::sdm_sensor_info>(device, (int)CURRENT_REQ);
+    auto voltage  = xrt_core::device_query<xq::sdm_sensor_info>(device, (int)VOLTAGE_REQ);
+    auto power  = xrt_core::device_query<xq::sdm_sensor_info>(device, (int)POWER_REQ);
+    if (current.empty() ^ voltage.empty() ^ power.empty()) {
+      sensor_array.push_back(std::make_pair("", pt.put("data_driven", false)));
+      is_data_driven = false;
+    }
+    else {
+      pt.put("data_driven", true);
+      std::string val;
+      for(size_t i = 0; i < current.size(); i++)
+      {
+        auto tmp = current[i];
+        pt.put("label", tmp.label);
+        pt.put("instantaneous", xrt_core::utils::format_base10_shiftdown3(tmp.input));
+        pt.put("max", xrt_core::utils::format_base10_shiftdown3(tmp.max));
+        pt.put("average", xrt_core::utils::format_base10_shiftdown3(tmp.average));
+        pt.put("highest", xrt_core::utils::format_base10_shiftdown3(tmp.highest));
+        pt.put("current.is_present", true);
+        pt.put("voltage.is_present", false);
+        pt.put("power.is_present", false);
```

Comment:
These are used in ReportElectrical to distinguish between voltage, power, current readings so that value extensions like "Amps", "Watts" etc added there. Please check this file once.